sql/pg_duckdb--0.0.1.sql

```diff
@@ -1,30 +1,97 @@
 LOAD 'pg_duckdb';
 
-CREATE OR REPLACE FUNCTION read_parquet(path text)
+CREATE OR REPLACE FUNCTION read_parquet(path text, binary_as_string BOOLEAN DEFAULT FALSE,
+                                                   filename BOOLEAN DEFAULT FALSE,
+                                                   file_row_number BOOLEAN DEFAULT FALSE,
+                                                   hive_partitioning BOOLEAN DEFAULT FALSE,
+                                                   union_by_name BOOLEAN DEFAULT FALSE)
 RETURNS SETOF record LANGUAGE 'plpgsql' AS
 $func$
 BEGIN
     RAISE EXCEPTION 'Function `read_parquet(TEXT)` only works with Duckdb execution.';
 END;
 $func$;
 
-CREATE OR REPLACE FUNCTION read_parquet(path text[])
+CREATE OR REPLACE FUNCTION read_parquet(path text[], binary_as_string BOOLEAN DEFAULT FALSE,
+                                                     filename BOOLEAN DEFAULT FALSE,
+                                                     file_row_number BOOLEAN DEFAULT FALSE,
+                                                     hive_partitioning BOOLEAN DEFAULT FALSE,
+                                                     union_by_name BOOLEAN DEFAULT FALSE)
 RETURNS SETOF record LANGUAGE 'plpgsql' AS
 $func$
 BEGIN
     RAISE EXCEPTION 'Function `read_parquet(TEXT[])` only works with Duckdb execution.';
 END;
 $func$;
 
-CREATE OR REPLACE FUNCTION read_csv(path text)
+
+-- Arguments 'columns' and 'nullstr' are currently not supported for read_csv
+
+CREATE OR REPLACE FUNCTION read_csv(path text, all_varchar BOOLEAN DEFAULT FALSE,
```

Comment:
I see `columns` is missing, is that because we can't pass the hash/JSON syntax?

```
 columns => {
        'FlightDate': 'DATE',
        'UniqueCarrier': 'VARCHAR',
        'OriginCityName': 'VARCHAR',
        'DestCityName': 'VARCHAR'
    }```

Reply:
Yes, not sure is there is postgresql way to provide struct for this option.

Reply:
Supporting this option isn't a big deal in itself, but it's a bit annoying to say "everything is supported except `columns`."

Supporting `struct_pack` is close but it wants bare key names. 

The [JSON type][1] does have a way to convert to struct, but the `::struct` part would fail in Postgres, and is very verbose.

```sql
SELECT '{"duck": 42}'::JSON::STRUCT(duck INTEGER);
```

@Tishj any ideas on how to get structs from Postgres into DuckDB?

[1]: https://duckdb.org/docs/extensions/json#json-type

Reply:
Depends on if this will autocast on the duckdb side, providing the struct as a string could let it cast if that's the case

Reply:
I think that closes to struct that we could have is to declare columns to be `JSON` type and argument as:

```
columns => '{"col1" : "type1", "col2" : "type2"}'
```

This is going to be received as `VARCHAR` type in `CSV extension` so probably we would need to cast/convert it to struct type (maybe do this only if postgresql mode is set #36).